pycvvdp/display_model.py

```diff
@@ -308,6 +331,11 @@ def forward( self, V ):
             L = pq2lin( V ).clip(0.005, self.Y_peak) + Y_black + Y_refl #TODO: soft clipping
         elif self.EOTF=='linear':
             L = V.clip(max(0.005, Y_black), self.Y_peak) + Y_refl #TODO: soft clipping
+        elif self.EOTF=='HLG':
+            gamma = 1.2
+            if self.Y_peak > 1000:
+                gamma = 1.2 + 0.42 * math.log10(self.Y_peak / 1000) - 0.07623 * math.log10(self.E_ambient / 5)
```

Comment:
@cosmin where does the correction term for ambient light come from? I do not see it in the rec 2100. Did I miss something?

Reply:
I took this from note 5f

![image](https://github.com/user-attachments/assets/b14eec26-2350-41dd-947b-0e67a1bedd28)

Reply:
@cosmin sorry, I wasn't entirely explicit -- I meant to ask about the correction term `- 0.07623 * math.log10(self.E_ambient / 5)`

Reply:
@krasuluk ah, so BT.2100 says

> Note 5g – The system gamma value may be decreased for brighter background and surround conditions.

But doesn't specify how to decrease the gamma for bright ambient conditions. However it turns out the BBC ran some experiments on this described in BBC Research & Development White Paper WHP 369 https://downloads.bbc.co.uk/rd/pubs/whp/whp-pdf-files/WHP369.pdf

Here's the relevant part

![image](https://github.com/user-attachments/assets/6b10a10c-27b4-4d2f-b07a-d1c146948159)

This is also captured in ITU-R BT.2390-10 which quotes the BBC experiment

> Recommendation ITU-R BT.2100 recognises that the HLG display gamma may need to be reduced in brighter viewing environments, to compensate for the differences in the adaptation state of the eye. The BBC conducted subjective tests to measure the change in gamma necessary to perceptually match images displayed across a range of peak luminances in the reference and in non-reference environments. Twenty-one viewers participated in the tests.

![image](https://github.com/user-attachments/assets/36a21df8-25b8-4bf4-9c9e-6ca759eb3cb9)

I kept the 0.07623 from the original BBC white paper, but we could round that to 0.076. 

BT.2390-10 also mentions another option 

![image](https://github.com/user-attachments/assets/7e98f023-c806-4583-beb3-b1c07bced92c)

from "Display of high dynamic range images under varying viewing conditions" http://dx.doi.org/10.1117/12.2274253 but I didn't investigate that further.

Reply:
Thanks so much for the detailed explanation and links @cosmin 🙏 all is clear now 👍